menu-shopping.js

```diff
@@ -33,66 +66,78 @@ class MenuShopping {
 				let ingredientCloudObj = new CB.CloudObject("MSIngredient");
 				let ingredientObj = new MSIngredient(name, description, ingredientCloudObj);
 				ingredientObj.save();
 
-				// Add it to the active appIngredents list
+				// Add it to the active appIngredients list
 				this.appIngredients.push(ingredientObj);
 
 				console.log(this.appIngredients);
 			}
-		});
+		});*/
 	}
 
-	listIngredients(tableIDIngredients) { 
-		console.log(this.appIngredients);
-
-		// see if the list of ingredients is already in the MenuShopping object
-		if (this.appIngredients.length == 0) {
-			console.log("calling the getAllDBIngredients function")
-			this.appIngredients = this.getAllDBIngredients();
-		}
-
+	displayList() {
 		// Now display the ingredient list in the table with the ID tableIDIngredients
 		console.log("in the list");
 		//clear the table and replace the headers
-		$(tableIDIngredients+" tbody").remove();
-		$(tableIDIngredients).append(`<tr><th>Name</th><th>Description</th></tr>`);
+		$("table#itemList tbody").remove();
+		$("table#itemList").append(`<tr><th>Name</th><th>Description</th></tr>`);
 
 		console.log(this.appIngredients);
 
 		for (let z=0; z<this.appIngredients.length; z++) {
 			console.log("Name: " + this.appIngredients[z].name + " Description: " + this.appIngredients[z].description);
-			$(tableIDIngredients).append(`<tr><td>${this.appIngredients[z]._name}</td><td>${this.appIngredients[z]._description}</td></tr>`);
+			$("table#itemList").append(`<tr><td>${this.appIngredients[z]._name}</td><td>${this.appIngredients[z]._description}</td></tr>`);
 		}
+	}
+
+	listIngredients() {
+		console.log(this.appIngredients);
 
+		// see if the list of ingredients is already in the MenuShopping object
+		if (this.appIngredients.length == 0) {
+			console.log("nothing to list")
+		} else {
+			this.displayList();
+		}
 	}
 
-	getAllDBIngredients() {
-		console.log("In-cache list is empty - loading from DB")
-		let ingredientList = [];
+	fetchIngredients() {
+		let deferred = $.Deferred();
+		this.getAllDBIngredients(deferred.resolve, this.setIngredients, this);
+		console.log("fetched ")
+		return deferred.promise();
+	}
 
-		let query = new CB.CloudQuery("MSIngredient");
-		query.exists('name');
+	setIngredients(list) {
+		console.log(list)
+		this.appIngredients = list;
+		console.log("in setIngredients. The appIngredients list is " + this.appIngredients)
+		this.listIngredients("table#itemList")
+	}
+
+  // typically we would pass in successCallback as a function,
+  // that's acutally how the cloudboost works right now
+	getAllDBIngredients() {
+		const query = new CB.CloudQuery("MSIngredient");
+		query.exists("name");
 		query.find({
-			success: function(list) {
+			success: (list) => {
 				// iterate through the list to add the items from the database to the object
-				console.log("getting items from the database - " + list.length + " items available")
-				for (let z=0; z<list.length; z++) {
```

Comment:
Instead of using the `for` loop you can use the `map` function. Map essentially loops through an array and replaces the object in it's position.

```js
const ary = [1,2,3,4,5].map(function(i) {
  return i + 1;
});

// the new array will now be [2,3,4,5,6];
```

Notice how I used es5 functions there, to do it with es6 you would write it like this.

```js
const ary = [1,2,3,4,5].map((i) => { return i + 1; });
```

Being that it's a single line you can actually reduce it further.

```js
const ary = [1,2,3,4,5].map((i) => i + 1);
```

You can remove the parentheses and return statement if it's a single line.